packages/core/src/tools/utils.ts

```diff
@@ -490,24 +498,13 @@ interface CircularReferenceChecker {
   hasAlreadyBeenSeen(value: any): boolean
 }
 function createCircularReferenceChecker(): CircularReferenceChecker {
-  if (typeof WeakSet !== 'undefined') {
-    const set: WeakSet<any> = new WeakSet()
-    return {
-      hasAlreadyBeenSeen(value) {
-        const has = set.has(value)
-        if (!has) {
-          set.add(value)
-        }
-        return has
-      },
-    }
-  }
-  const array: any[] = []
+  // Using a weakmap instead of a weakset to support IE11
+  const map: WeakMap<any, boolean> = new WeakMap()
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏great, love it :)